pkg/block/metrics.go

```diff
@@ -134,3 +137,49 @@ func (m *MetricsAdapter) GetRegion(ctx context.Context, storageID, storageNamesp
 func (m *MetricsAdapter) RuntimeStats() map[string]string {
 	return m.adapter.RuntimeStats()
 }
+
+type Histograms struct {
+	durationHistograms    *prometheus.HistogramVec
+	requestSizeHistograms *prometheus.HistogramVec
+	adapterStatsID        *string
+}
+
+func BuildHistogramsInstance(name string, adapterStatsID *string) Histograms {
+	labelNames := []string{"operation", "error"}
+	if adapterStatsID != nil {
+		labelNames = append(labelNames, "adapter_stats_id")
+	}
+	var durationHistograms = promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name: name + "_operation_duration_seconds",
+			Help: "durations of outgoing " + name + " operations",
+		},
+		labelNames,
+	)
```

Comment:
I believe there is an issue here, each histogram can be initiated one time per name, otherwise it will panic. This means that:
1. We can't initiate a block adapter twice!
2. Don't think this can be used for 2 different adapterStatesIDs

Reply:
OK, I see there's a nil check to ensure initialization happens only once. However, I still believe we should prioritize registering the metric during package-level initialization rather than run time